core/new-gui/src/app/workspace/service/autocomplete/model/autocomplete.service.ts

```diff
@@ -0,0 +1,175 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+
+import { AppSettings } from '../../../../common/app-setting';
+import { SourceTableNamesAPIResponse, SuccessExecutionResult } from '../../../types/autocomplete.interface';
+
+import { AutocompleteUtils } from '../util/autocomplete.utils';
+import { OperatorPredicate } from '../../../types/workflow-common.interface';
+import { OperatorMetadata, OperatorSchema } from '../../../types/operator-schema.interface';
+import { OperatorMetadataService } from '../../operator-metadata/operator-metadata.service';
+import { WorkflowActionService } from '../../workflow-graph/model/workflow-action.service';
+import { Observable } from 'rxjs/Observable';
+import '../../../../common/rxjs-operators';
+
+import { combineLatest } from 'rxjs/observable/combineLatest';
+import { Subject } from 'rxjs/Subject';
+import { JSONSchema4 } from 'json-schema';
+import { merge } from 'rxjs/observable/merge';
+
+export const SOURCE_TABLE_NAMES_ENDPOINT = 'resources/table-metadata';
+export const AUTOMATED_SCHEMA_PROPAGATION_ENDPOINT = 'queryplan/autocomplete';
+
+/**
+ * Autocomplete service is used for the purpose of automated schema propagation. The service does
+ * following tasks: -
+ *  1. Originally the source operators have tableName property which has no options. This service fetches
+ *    source table names from backend and adds them as an enum array to the tableName property.
+ *  2. The autocomplete API at backend is invoked using invokeSchemaPropagationAPI() method.
+ */
+@Injectable()
+export class AutocompleteService {
+  // the input schema of operators in the current workflow as returned by the autocomplete API
+  public operatorInputSchemaMap: JSONSchema4 = {};
+
+  public autocompleteAPIExecutedStream = new Subject<string>();
```

Comment:
Don't make the `Subject` variable directly public, because other components can also call `subject.next` and push things to the subject. Set it to `private` instead. 
I see there's only a test case in property editor component that calls `next` on this stream, we can use other methods to write the test case.